internal/validate/install/install.go

```diff
@@ -71,6 +99,30 @@ func Validate(ctx context.Context, client api.Client, config *ValidationSpec) er
 	return nil
 }
 
+func checkExecutors(ctx context.Context, client api.Client, query string, variables map[string]interface{}) (int, error) {
+	q := clientQuery{
+		opName:    "CheckExecutorConnection",
+		query:     query,
+		variables: variables,
+	}
+
+	var result struct {
+		Executor struct {
+			TotalCount int `json:"totalCount"`
+		} `json:"executors"`
+	}
+
+	ok, err := client.NewRequest(q.query, q.variables).Do(ctx, &result)
+	if err != nil {
+		return 0, errors.Wrap(err, "checkExecutors failed")
```

Comment:
Consider using `-1` as the int value when returning an error. See note [here](https://chortle.ccsu.edu/qbasic/chapter09/bc09_15.html#:~:text=A%20sentinel%20value%20should%20be,it%20without%20changing%20the%20sum.) on sentinel values. Just a preference, but there's lots of precedence for this convention. Also, you do, in fact, validly return `0` at times. -1 is a value you would never expect to receive if the function is behaving as expected.